### PR TITLE
Enhances Roundstart Antagonist Selection yadda yadda yadda

### PR DESCRIPTION
**Pull request by rbertoche**
I noticed upstream antag selection is still iffy so there you go.

You already made it slightly better but this makes ChooseAntags significantly better.

No public API changes besides adding stuff, to not cause breakage.

Waiting for the A. I. to spot all my flaws and condemn me to hell.

The same as this, adapted to goob:

https://github.com/Simple-Station/Einstein-Engines/pull/1339

No testing made. At least I don't lie about it.

Cheers

## Requirements

- [ ] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [ ] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

:cl:
- tweak: Enhances roundstart antagonist selection making sure all slots are allocated

### CODE REVIEW

File: Content.Server/Antag/AntagSelectionSystem.cs [hunk 1]
```diff
@@ -238,26 +238,68 @@ public void ChooseAntags(Entity<AntagSelectionComponent> ent,
...
                 {
-                    Log.Warning($"Somehow picked {session} for an antag when this rule already selected them previously");
-                    continue;
-                }
+                    return !ent.Comp.SelectedSessions.Contains(session_) &&
+                        !failed.Contains(session_);
+                });
+                retry++;
             }
+        }
```
[the remark this answers is not in the document]
Comment (rbertoche) [reply Content.Server/Antag/AntagSelectionSystem.cs:290]: It is in a constant dumb A. I.

Ok this is resolved, it will work with negatives.

**Comment by rbertoche** [comment]
May or may not be useful for this, #1142 the author of bloodcult was complaining about not enough antags getting assigned.

**Comment by rbertoche** [comment]
Am fixing the broken test TryStopNukeOpsFromConstantlyFailing

**Comment by rbertoche** [comment]
That test is fixed, now lets see of it didn't break something else.

**Comment by Piras314** [comment]
I'll test this before merging... eventually. 'Tis the season to be busy.

**Comment by Piras314** [comment]
john station did it

**Pull request closed without merging**